msgpack/fallback.py

```diff
@@ -246,16 +277,13 @@ def __init__(self, file_like=None, read_size=0, use_list=True,
             raise TypeError("`ext_hook` is not callable")
 
     def feed(self, next_bytes):
-        if isinstance(next_bytes, array.array):
-            next_bytes = next_bytes.tostring()
-        elif isinstance(next_bytes, bytearray):
-            next_bytes = bytes(next_bytes)
         assert self._fb_feeding
-        if (self._fb_buf_n + len(next_bytes) - self._fb_sloppiness
-                        > self._max_buffer_size):
+        view = _get_data_from_buffer(next_bytes, 1)
+        L = len(view)
+        if self._fb_buf_n + L - self._fb_sloppiness > self._max_buffer_size:
             raise BufferFull
-        self._fb_buf_n += len(next_bytes)
-        self._fb_buffers.append(next_bytes)
+        self._fb_buf_n += L
+        self._fb_buffers.append(view)
```

Comment:
Since we will be copying anyway it would make sense to use a single, larger buffer. I experimented in that direction, but it was quite complex.